torchrl/data/datasets/openx.py

```diff
@@ -572,10 +572,12 @@ def __iter__(self):
             else:
                 yield data
 
-    def get(self, index: int) -> Any:
+    def get(self, index: range) -> Any:
         if not isinstance(index, range):
-            # we use a range to indicate how much data we want
-            raise RuntimeError("iterable datasets do not support indexing.")
+            if (index[1:] != index[:-1] + 1).any():
+                # we use a range to indicate how much data we want
+                raise RuntimeError("iterable datasets do not support indexing.")
+            index = range(index.shape[0])
```

Comment:
This is a bit expensive but it's necessary since we can't be sure of the type of the input